### ISSUE TITLE
datamodel optional size parameter

### ISSUE THREAD

**Issue by rcabanasdepaz** [PGM-Lab/InferPy#114]
When defining a model, allow to directly specify the extended model. This can be done with an optional parameter:

```
@models.probmodel
def model():
    w = models.Normal(np.zeros([4,2], np.float32), 1., name='w')
    with models.datamodel(size=100):
        x = models.Normal(np.zeros([2,4], np.float32), 1., name='x')
        y = models.Normal(tf.constant(10.) + tf.matmul(x, w), 0.1, name='y')
```   

This might make easier the model definition, as the user can "play" with the operations involving the extended rvs.

**Comment by jcozar87** [comment]
As long as I understand, to _play_ means executing the code inside the _probmodel_ function sequentially in the interpreter.

The problem of doing that, is that _datamodel_ **requires** the internal `prob_model` context builder, to check the dependencies among Random Variables.

This behaviour is limited to `ProbModel` objects to allow users to create Random Variables and play with them with no limits (but always outside a probmodel context).

Therefore, I suggest to close this issue, at least until a new use case is detected (but always to be used inside a probmodel).

**Comment by jcozar87** [comment]
Should be implemented after #119
